### PR TITLE
EdinburghGenomics/Analysis-Driver#180: Adding clarity.get_library_id.

### PR DESCRIPTION
**Pull request by mwhamgenomics**
This adds a function to `clarity`, allowing it to query library IDs. Required for EdinburghGenomics/Analysis-Driver#180.

### CODE REVIEW

File: egcg_core/clarity.py [hunk 1]
```diff
@@ -295,9 +293,21 @@ def get_sample_release_date(sample_id):
...
-    return procs[0].date_run
+    artifacts = connection().get_artifacts(
+        sample_name=sample_id,
+        type='Analyte',
+        process_type='AUTOMATED - Clean Up ALP'
+    )
+    if len(artifacts) == 1:
+        return artifacts[0].udf.get('Raw Library ID')
+    else:
+        app_logger.warning("%s 'Clean Up ALP' Artifacts found for sample %s", (len(artifacts), sample_id))
```
Comment (tcezard) [inline egcg_core/clarity.py]: Could you raise an exception instead of a warning?

**Comment by tcezard** [comment]
As discussed yesterday. The idea I had about this function was wrong. We do not want to assume that only one library exist for any given sample. 
We can keep this function but search for all libraries but will have to implement another function that will search for a library based on the run it was on.

File: egcg_core/clarity.py [hunk 2]
```diff
@@ -295,9 +293,21 @@ def get_sample_release_date(sample_id):
...
-    return procs[0].date_run
+    artifacts = connection().get_artifacts(
+        sample_name=sample_id,
+        type='Analyte',
+        process_type='AUTOMATED - Clean Up ALP'
+    )
+    if not artifacts:
+        app_logger.warning("No 'Clean Up ALP' artifacts found for sample " + sample_id)
+    else:
+        return set(a.udf.get('Raw Library ID') for a in artifacts)
```
Comment (mwhamgenomics) [inline egcg_core/clarity.py:313]: The `Raw Library ID` UDF is not actually a library ID - it's actually a container name that is shared across all samples in the plate. We need to decide what we are going to use as the library ID, then query for that, and that should give us a list of correct library IDs.

We need a new function called, e.g, `get_library_id_for_run_element`, which can go from a run element to a real library ID.

**Comment by tcezard** [comment]
Won't happen any time soon. closing for now

**Pull request closed without merging**